src/client/Microsoft.Identity.Client/Extensions.cs

```diff
@@ -0,0 +1,101 @@
+
+        }
+
+        /// <summary>
+        /// Returns true if MSAL can use an embedded webview (browser). 
+        /// </summary>
+        /// <remarks>
+        /// Currently there are no embedded webviews on Mac and Linux. On Windows, app developers or users should install 
+        /// the WebView2 runtime and this property will inform if the runtime is available, see https://aka.ms/msal-net-webview2
+        /// </remarks>
+        public static bool IsEmbeddedWebViewAvailable(this IPublicClientApplication publicClientApplication)
+        {
+            if (publicClientApplication is PublicClientApplication pca)
+            {
+                return pca.IsEmbeddedWebViewAvailable();
+            }
+
+            throw new ArgumentException("This extension method is only available for the PublicClientApplication implementation " +
+                "of the IPublicClientApplication interface.");
+        }
+
+        /// <summary>
+        /// Returns false when the program runs in headless OS, for example when SSH-ed into a Linux machine.
+        /// Browsers (webviews) and brokers cannot be used if there is no UI support. Instead, please use <see cref="PublicClientApplication.AcquireTokenWithDeviceCode(IEnumerable{string}, Func{DeviceCodeResult, Task})"/>
+        /// </summary>
+        public static bool IsDesktopSession(this IPublicClientApplication publicClientApplication)
+        {
+            if (publicClientApplication is PublicClientApplication pca)
+            {
+                return pca.IsDesktopSession();
+            }
+
+            throw new ArgumentException("This extension method is only available for the PublicClientApplication implementation " +
+                "of the IPublicClientApplication interface.");
+        }
+
+        /// <summary>
+        /// Returns true if a broker can be used. 
+        /// </summary>
+        /// <remarks>
+        /// On Windows, the broker (WAM) can be used on Win10 and is always installed.
+        /// On Mac, Linux and older versions of Windows a broker is not available.
+        /// On mobile, the device must be Intune joined and Authenticator or Company Portal must be installed.
+        /// </remarks>
+        public static bool IsBrokerAvailable(this IPublicClientApplication publicClientApplication)
+        {
+            if (publicClientApplication is PublicClientApplication pca)
+            {
+                return pca.IsBrokerAvailable();
+            }
+
+            throw new ArgumentException("This extension method is only available for the PublicClientApplication implementation " +
+                "of the IPublicClientApplication interface.");
+        }
+
+        /// <summary>
+        /// Returns the certificate used to create this <see cref="ConfidentialClientApplication"/>, if any.
+        /// </summary>
+        public static X509Certificate2 GetCertificate(this IConfidentialClientApplication confidentialClientApplication)
```

Comment:
Fixes https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2120 without a breaking change

Reply:
any value in having extensions for pca and extensions for cca separately?

---
In reply to: [589415189](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2458#discussion_r589415189) [](ancestors = 589415189)

Reply:
Thank you for looking up the work item @jmprieur , apreciate it :). I remembered the problem, but forgot we were still tracking it....

Reply:
@jennyf19 - I can't think of any.

Reply:
@bgavrilMS i meant more of if we ever try to split the library again, cca/pca, then we'd have separate extensions for both already.